test-harness/infratests/validate.go

```diff
@@ -0,0 +1,128 @@
+/*
+Package infratests This file provides validation utilities that can be used by the core testing constructs
+*/
+package infratests
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// This function validates that a set of search targets exist in a map. The intended use case is to allow
+// a user of this library to provide a set of *expected values* that should exist within a larger set of
+// *actual values.* In other words, this is a map equality check that does not care about ordering in keys/lists
+// and will not fail if the *expected values* are a subset of the *actual values.*
+//
+// In the case that the *expected values* are found in the *actual values,* nil will be returned. Otherwise an
+// error describing the mismatch will be returned.
+//
+// Some examples (in pseudo-code):
+//	- Exact Match:
+//		a := {"key1":[{"key2" : "foo"}, {"key2" : "bar"}]}
+//		verifyTargetsExistInMap(a, a) --> nil
+//
+//	- Subset Match #1:
+//		a := {"key1":[{"key2" : "foo"}, {"key2" : "bar"}]}
+//		b := {"key1":[{"key2" : "foo"}]}
+//		verifyTargetsExistInMap(a, b) --> nil
+//
+//	- Subset Match #2:
+//		a := {"key1":[{"key2" : "foo"}, {"key2" : "bar"}]}
+//		b := {"key1":[]}
+//		verifyTargetsExistInMap(a, b) --> nil
+//
+//	- Subset Match #3:
+//		a := {"key1":{"key2": "foo", "key3", "bar"}}
+//		b := {"key1":{"key3", "bar"}}
+//		verifyTargetsExistInMap(a, b) --> nil
+//
+//	- Mismatch #1:
+//		a := {"key1":{"key2": "foo", "key3", "bar"}}
+//		b := {"key1":[]}
+//		verifyTargetsExistInMap(a, b) --> ERROR: wrong type for key `key1`
+//
+//	- Mismatch #2:
+//		a := {"key1":{"key2": "foo", "key3", "bar"}}
+//		b := {"key1":{"key4": "foo"}}
+//		verifyTargetsExistInMap(a, b) --> ERROR: `key4` not found
+//
+//	- Mismatch #3:
+//		a := {"key1":{"key2": "foo", "key3", "bar"}}
+//		b := {"key1":{"key2": "bar"}}
+//		verifyTargetsExistInMap(a, b) --> ERROR: wrong value for `key2`
+//
+// The algorithm used to do the equality check is to execute a BFS search in parallel for both maps. In the case
+// that a mismatch (the values do not match, or the shape of the maps is different in a way that indicates a mismatch)
+// is identified, the search will be terminated. If no mismatches are found then `nil` is returned
+func verifyTargetsExistInMap(dataSource map[string]interface{}, searchTargets map[string]interface{}) error {
+	for targetKey := range searchTargets {
+		candidateMatch := dataSource[targetKey]
+		target := searchTargets[targetKey]
+
+		// both maps should contain the target search key
+		if candidateMatch == nil || target == nil {
```

Comment:
```suggestion
		if !candidateMatchExist || !targetExist {
```

Reply:
Will go ahead and add this. I had it before but took it out in an effort to make things more concise. I think the approach you suggested adds clarity, though, as it does not rely on the "zero" value being `nil` for `interface{}`